extensions/common/gcp/gcp-core/src/main/java/org/eclipse/dataspaceconnector/gcp/core/storage/StorageServiceImpl.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2022 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LCC - Initial implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.gcp.core.storage;
+
+import com.google.cloud.Binding;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import org.eclipse.dataspaceconnector.gcp.core.common.BucketWrapper;
+import org.eclipse.dataspaceconnector.gcp.core.common.GcpException;
+import org.eclipse.dataspaceconnector.gcp.core.common.ServiceAccountWrapper;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+
+import java.util.ArrayList;
+
+public class StorageServiceImpl implements StorageService {
+
+    private final Storage storageClient;
+    private final Monitor monitor;
+
+
+    public StorageServiceImpl(Storage storageClient, Monitor monitor) {
+        this.storageClient = storageClient;
+        this.monitor = monitor;
+    }
+
+    @Override
+    public BucketWrapper getOrCreateEmptyBucket(String bucketName, String location) {
+        var bucket = storageClient.get(bucketName);
+        if (bucket == null) {
+            monitor.info("Creating new bucket " + bucketName);
+            bucket = storageClient.create(BucketInfo.newBuilder(bucketName).setLocation(location).build());
```

Comment:
Please avoid variable overwriting, `Optional` could be used to better separate concerns (get or create bucket, validate it, return wrapper):
```java
        var bucket = Optional.of(storageClient)
                .map(it -> it.get(bucketName))
                .orElseGet(() -> {
                    monitor.info("Creating new bucket " + bucketName);
                    return storageClient.create(BucketInfo.newBuilder(bucketName).setLocation(location).build());
                });

        if (!bucket.getLocation().equals(location)) {
            throw new GcpException("Bucket " + bucketName + " already exists but in wrong location");
        }
        
        return new BucketWrapper(bucket.getName());
```

Reply:
done